pages/accessibilitysupport.php

```diff
@@ -0,0 +1,160 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Theme Boost Union - Accessibility support page.
+ *
+ * @package    theme_boost_union
+ * @copyright  2024 Katalin Lukacs Toth, ZHAW Zurich University of Applied Sciences <lukc@zhaw.ch>
+ * @copyright  2024 Simon Schoenenberger, ZHAW Zurich University of Applied Sciences <scgo@zhaw.ch>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+// Include config.php.
+// @codingStandardsIgnoreStart
+// Let codechecker ignore the next line because otherwise it would complain about a missing login check
+// after requiring config.php which is really not needed.
+require(__DIR__ . '/../../../config.php');
+// @codingStandardsIgnoreEnd
+
+// Require the necessary libraries.
+require_once($CFG->dirroot.'/theme/boost_union/lib.php');
+require_once($CFG->dirroot.'/theme/boost_union/locallib.php');
+
+// Set page URL.
+$PAGE->set_url('/theme/boost_union/pages/accessibilitysupport.php');
+
+// Set page layout.
+$PAGE->set_pagelayout('standard');
+
+// Set page context.
+$PAGE->set_context(context_system::instance());
+
+// Add page name as body class.
+$PAGE->add_body_class('theme_boost_union-accessibilitysupport');
+
+// Get theme config.
+$config = get_config('theme_boost_union');
+
+// If the Accessibility supportpage is disabled, we just show a short friendly warning page and are done.
+if ($config->enableaccessibilitysupport != THEME_BOOST_UNION_SETTING_SELECT_YES) {
+    echo $OUTPUT->header();
+    $notification = new \core\output\notification(get_string('accessibilitysupportdisabled', 'theme_boost_union'),
+            \core\output\notification::NOTIFY_INFO);
+    $notification->set_show_closebutton(false);
+    echo $OUTPUT->render($notification);
+    echo $OUTPUT->footer();
+    die;
+}
+
+// If user login is required, we redirect to the login page.
+if ($config->enableaccessibilitysupportwithoutlogin != THEME_BOOST_UNION_SETTING_SELECT_YES) {
+    if (!isloggedin() || isguestuser()) {
+        redirect(get_login_url());
+    }
+}
+
+// Set page title.
+$PAGE->set_title(theme_boost_union_get_staticpage_pagetitle('accessibilitysupport'));
+
+$form = new \theme_boost_union\form\accessibilitysupport_form();
+
+// Get the page where we came from.
+$referrerpage = $form->get_referrer_page();
+if (!$referrerpage) {
+    $referrerpage = $CFG->wwwroot;
+}
+
+if ($form->is_cancelled()) {
+    redirect($referrerpage);
+} else if ($form->is_submitted() && $form->is_validated() && confirm_sesskey()) {
+    $data = $form->get_data();
+
+    // Do not send system information when not allowed.
+    if (!$data->sendtechinfo) {
+        unset($data->techinfo);
+    }
+
```

Comment:
If you know the feature, and the page this is straight forward and good,
but when you are coming here later - e.g. for bug research - it would be nice to have some more comments for guidance:

- To clarify intent during the form data processing, please add some comments to the if/else branches.